tests/unit/config/config_test.py

```diff
@@ -231,6 +231,19 @@ def test_named_volume_config_empty(self):
         assert volumes['simple'] == {}
         assert volumes['other'] == {}
 
+    def test_volume_invalid_driver_opt(self):
+        config_details = build_config_details({
+            'version': '2',
+            'services': {
+                'simple': {'image': 'busybox'}
+            },
+            'volumes': {
+                'simple': {'driver_opts': {'size': 42}},
+            }
+        })
+        with self.assertRaises(ConfigurationError):
+            config.load(config_details)
```

Comment:
Minor: we've been using `pytest.raises(...)` in most places. It's basically the same idea, but returns a different object which I think makes it easier to inspect errors. It will also let us transition off of `unittest.TestCase` classes.

Probably worth checking the error message here as well, since the error itself can be raised  for many different conditions.